nipype/interfaces/base/traits_extension.py

```diff
@@ -54,244 +95,239 @@ class Str(Unicode):
-                 value='',
-                 filter=None,
-                 auto_set=False,
-                 entries=0,
-                 exists=False,
-                 **metadata):
-        """ Creates a File trait.
-
-        Parameters
-        ----------
-        value : string
-            The default value for the trait
-        filter : string
-            A wildcard string to filter filenames in the file dialog box used by
-            the attribute trait editor.
-        auto_set : boolean
-            Indicates whether the file editor updates the trait value after
-            every key stroke.
-        exists : boolean
-            Indicates whether the trait value must be an existing file or
-            not.
-
-        Default Value
-        -------------
-        *value* or ''
-        """
-        self.filter = filter
-        self.auto_set = auto_set
-        self.entries = entries
+    info_text = 'a pathlike object or string'
+    exists = False
+    pathlike = False
+    resolve = False
+    _is_file = False
+    _is_dir = False
+
+    def __init__(self, value=Undefined,
+                 exists=False, pathlike=False, resolve=False, **metadata):
+        """Create a BasePath trait."""
         self.exists = exists
+        self.resolve = resolve
+        self.pathlike = pathlike
+        if any((exists, self._is_file, self._is_dir)):
+            self.info_text += ' representing a'
+            if exists:
+                self.info_text += 'n existing'
+            if self._is_file:
+                self.info_text += ' file'
+            elif self._is_dir:
+                self.info_text += ' directory'
+            else:
+                self.info_text += ' file or directory'
 
-        if exists:
-            self.info_text = 'an existing file name'
+        super(BasePath, self).__init__(value, **metadata)
 
-        super(File, self).__init__(value, **metadata)
+    def validate(self, objekt, name, value, return_pathlike=False):
+        """Validate a value change."""
+        try:
+            if USING_PATHLIB2 and isinstance(value, newstr):
+                value = '%s' % value  # pathlib2 doesn't like newstr
+            value = Path(value)  # Use pathlib's validation
+        except Exception:
+            self.error(objekt, name, str(value))
 
-    def validate(self, object, name, value):
-        """ Validates that a specified value is valid for this trait."""
-        validated_value = super(File, self).validate(object, name, value)
-        if not self.exists:
-            return validated_value
-        elif os.path.isfile(value):
-            return validated_value
-        else:
-            raise TraitError(
-                args='The trait \'{}\' of {} instance is {}, but the path '
-                ' \'{}\' does not exist.'.format(name, class_of(object),
-                                                 self.info_text, value))
+        if self.exists:
+            if self.exists and not value.exists():
```

Comment:
Already covered by previous line.

```suggestion
            if not value.exists():
```